projects/packages/blaze/src/class-dashboard-rest-controller.php

```diff
@@ -52,6 +52,17 @@ public function register_rest_routes() {
 			)
 		);
 
+		// WordAds DSP API Posts routes
+		register_rest_route(
+			static::$namespace,
+			sprintf( '/sites/%d/wordads/dsp/api/v1/wpcom/sites/%d/blaze/posts(\?.*)?', $site_id, $site_id ),
```

Comment:
That is to make the regex works for sites using plain permalinks. I am not storing that regex patch in a variable.
The actual parse of the parameters happens on the function `build_subpath_with_query_strings`, and in there, we call the `stripslashes_deep` method to clean it up.

For that specific method, the query params that we expect are for the page, title, sorting, etc.

Reply:
Maybe it's a config problem, but in the last PR I had to append `(\?.*)?` that to the end of the regex, because for sites using plain permalinks, all the request from the client that had query string weren't working.

That was because the path that the controller was matching against, had the query param on it (this doesn't happen for sites using the other configurations).

If there is a better way to handle that, let me know. I can remove that extra `(\?.*)?` and leave it like it was before.

Reply:
Let's address this in a follow-up PR. 👍